keygen/src/keygen.rs

```diff
@@ -41,6 +41,86 @@ struct GrindMatch {
     count: AtomicU64,
 }
 
+const WORD_COUNT_ARG: ArgConstant<'static> = ArgConstant {
+    long: "word-count",
+    name: "word_count",
+    help: "Specify the number of words that will be present in the generated seed phrase",
+};
+
+const LANGUAGE_ARG: ArgConstant<'static> = ArgConstant {
+    long: "language",
+    name: "language",
+    help: "Specify the mnemonic lanaguage that will be present in the generated seed phrase",
+};
+
+const NO_PASSPHRASE_ARG: ArgConstant<'static> = ArgConstant {
+    long: "no-bip39-passphrase",
+    name: "no-passphrase",
+    help: "Do not prompt for a BIP39 passphrase",
+};
+
+const NO_OUTFILE_ARG: ArgConstant<'static> = ArgConstant {
+    long: "no-outfile",
+    name: "no_outfile",
+    help: "Only print a seed phrase and pubkey. Do not output a keypair file",
+};
+
+fn word_count_arg<'a, 'b>() -> Arg<'a, 'b> {
+    Arg::with_name(WORD_COUNT_ARG.name)
+        .long(WORD_COUNT_ARG.long)
+        .possible_values(&["12", "15", "18", "21", "24"])
+        .default_value("12")
+        .value_name("NUMBER")
+        .takes_value(true)
+        .help(WORD_COUNT_ARG.help)
+}
+
+fn language_arg<'a, 'b>() -> Arg<'a, 'b> {
+    Arg::with_name(LANGUAGE_ARG.name)
+        .long(LANGUAGE_ARG.long)
+        .possible_values(&[
+            "english",
+            "chinese-simplified",
+            "chinese-traditional",
+            "japanese",
+            "spanish",
+            "korean",
+            "french",
+            "italian",
+        ])
+        .default_value("english")
+        .value_name("LANGUAGE")
+        .takes_value(true)
+        .help(LANGUAGE_ARG.help)
+}
+
+fn no_passphrase_arg<'a, 'b>() -> Arg<'a, 'b> {
+    Arg::with_name(NO_PASSPHRASE_ARG.name)
+        .long(NO_PASSPHRASE_ARG.long)
+        .alias("no-passphrase")
+        .help(NO_PASSPHRASE_ARG.help)
+}
+
+fn no_outfile_arg<'a, 'b>() -> Arg<'a, 'b> {
+    Arg::with_name(NO_OUTFILE_ARG.name)
+        .long(NO_OUTFILE_ARG.long)
+        .conflicts_with_all(&["outfile", "silent"])
+        .help(NO_OUTFILE_ARG.help)
+}
+
+trait NewGrindCommonArgs {
+    fn new_grind_common_args(self) -> Self;
```

Comment:
Since these are used in both `new` and `grind`, can you call this trait and the method something more generic?

Reply:
Well "NewGrindCommonArgs" was intended to represent that it was for "new" and "grind", but I can change this to "NewAndGrindCommonArgs" if that's clearer.

That's pretty ugly though.  I will instead choose something semantic like "KeyGenerationCommonArgs" since new and grind fundamentally both generate keys.

Reply:
Oh I'm a dork; I read that as new modifying grind 🤦‍♀️ 
I still prefer `KeyGenerationCommonArgs` though

Reply:
You're not a dork, I can totally see where that was confusing.  I changed it to KeyGenerationCommonArgs.  Thanks for your feedback.